parser/tavor.go

```diff
@@ -1565,102 +1564,33 @@ func (p *tavorParser) parseTypedTokenDefinition(variableScope *token.VariableSco
 		return zeroRune, err
 	}
 
-	var tok token.Token
-	usedArguments := make(map[string]struct{})
-
-	switch typ {
-	case "Int":
-		rawFrom, okFrom := arguments["from"]
-		rawTo, okTo := arguments["to"]
-
-		from := 0
-		if okFrom {
-			from, err = strconv.Atoi(rawFrom)
-			if err != nil {
-				return zeroRune, &token.ParserError{
-					Message:  "\"from\" needs an integer value",
-					Type:     token.ParseErrorInvalidArgumentValue,
-					Position: p.scan.Pos(),
-				}
-			}
-		}
-
-		to := math.MaxInt32
-		if okTo {
-			to, err = strconv.Atoi(rawTo)
-			if err != nil {
-				return zeroRune, &token.ParserError{
-					Message:  "\"to\" needs an integer value",
-					Type:     token.ParseErrorInvalidArgumentValue,
-					Position: p.scan.Pos(),
-				}
-			}
-		}
-
-		step := 1
-
-		if raw, ok := arguments["step"]; ok {
-			step, err = strconv.Atoi(raw)
-			if err != nil {
-				return zeroRune, &token.ParserError{
-					Message:  "\"step\" needs an integer value",
-					Type:     token.ParseErrorInvalidArgumentValue,
-					Position: p.scan.Pos(),
-				}
-			}
-
-			usedArguments["step"] = struct{}{}
-		}
-
-		usedArguments["from"] = struct{}{}
-		usedArguments["to"] = struct{}{}
-
-		tok = primitives.NewRangeIntWithStep(from, to, step)
-	case "Sequence":
-		start := 1
-		step := 1
-
-		if raw, ok := arguments["start"]; ok {
-			start, err = strconv.Atoi(raw)
-			if err != nil {
-				return zeroRune, &token.ParserError{
-					Message:  "\"start\" needs an integer value",
-					Type:     token.ParseErrorInvalidArgumentValue,
-					Position: p.scan.Pos(),
-				}
-			}
-		}
-
-		if raw, ok := arguments["step"]; ok {
-			step, err = strconv.Atoi(raw)
-			if err != nil {
-				return zeroRune, &token.ParserError{
-					Message:  "\"step\" needs an integer value",
-					Type:     token.ParseErrorInvalidArgumentValue,
-					Position: p.scan.Pos(),
-				}
-			}
+	// find the constructor for the current type
+	createTok, found := token.GetTypedConstructor(typ)
+	if found != nil {
+		return zeroRune, &token.ParserError{
+			Message:  found.Error(),
+			Type:     token.ParseErrorUnknownTypedTokenType,
+			Position: p.scan.Pos(),
 		}
+	}
 
-		usedArguments["start"] = struct{}{}
-		usedArguments["step"] = struct{}{}
-
-		tok = sequences.NewSequence(start, step)
-	default:
+	// construct the typed token
+	argParser := argumentsParser{arguments, make(map[string]struct{}), nil}
```

Comment:
Please put the initialization in a "newArgumentsParser" function and use the field names instead of doing this in one line. I know this is very verbose but it is easier to maintain.